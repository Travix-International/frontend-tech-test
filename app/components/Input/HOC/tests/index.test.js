import React from 'react';
import { create } from 'react-test-renderer';

import InputHOC from '../index';

const Component = InputHOC('Component');
const initial = {};
const renderComponent = props => {
  const finalProps = { ...initial, ...props };

  return <Component {...finalProps} />;
};

describe('<InputHOC />', () => {
  it('should render component', () => {
    const tree = create(renderComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should call onChange when input changes', () => {
    const name = 'name';
    const value = 'value';
    const onChange = jest.fn();
    const props = {
      name,
      onChange
    };
    const component = create(renderComponent(props));
    const instance = component.getInstance();

    instance.handleChange({ target: { value } });

    expect(onChange).toHaveBeenCalledWith(value, name);
  });
});
