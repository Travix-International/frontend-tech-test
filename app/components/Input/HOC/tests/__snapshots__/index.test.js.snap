// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputHOC /> should render component 1`] = `
<Component
  onChange={[Function]}
/>
`;
